python/flexflow/torch/model.py

```diff
@@ -1417,24 +1418,62 @@ def is_unsqueeze(slice_elem):
             """Returns if the slice is equivalent to unsqueezing that
             dimension."""
             return slice_elem is None
+
+        def is_truncate(slice_elem, old_size):
+            start = 0 if slice_elem.start == None else slice_elem.start
+            new_size = slice_elem.stop - start
+            return new_size < old_size
+        
+        def is_single_element(slice_elem):
+            return type(slice_elem) == int
+
         shape = tensor.dims
-        # Match dimensions from right to left
-        new_shape = []  # append then reverse
+
+        # Fewer slices than input dimensions
+        diff = len(shape) - len(slices)
+        if diff > 0:
+            slices_lst = list(slices)
+            for i in range(diff):
+                slices_lst.append(slice(None, None, None))
+            slices = tuple(slices_lst)
+
+        # Match dimensions from right to left                                                                  
+        new_shape = []  # append then reverse                                                                  
         j = len(shape) - 1
+        curr_tensor = copy.copy(tensor)
+
         for slice_elem in reversed(slices):
             if is_colon(slice_elem):
                 assert j >= 0
                 new_shape.append(shape[j])
                 j -= 1
             elif is_unsqueeze(slice_elem):
                 new_shape.append(1)
+            elif is_single_element(slice_elem):
+                assert j >= 0
+                splits = [1, shape[j] - 1]
+                curr_tensor = ffmodel.split(input=curr_tensor, sizes=splits, axis=j, name=name)[0]
```

Comment:
It might be nice to create two names in this case. Names are currently used for debugging I think, so having two operators with the same name seems confusing. Do people with more python frontend experience than me agree? @eddy16112 @gabrieleoliaro @jiazhihao

Reply:
Discussion about @lockshaw's question: https://github.com/flexflow/FlexFlow/discussions/491